### ISSUE TITLE
ISSUE

### ISSUE THREAD

**Issue by alfielytorres** [gjtorikian/commonmarker#82]
Fetching gem metadata from https://rubygems.org/.........
Using concurrent-ruby 1.0.5
Using i18n 0.9.5
Using minitest 5.11.3
Using thread_safe 0.3.6
Using tzinfo 1.2.5
Using activesupport 4.2.10
Using public_suffix 2.0.5
Using addressable 2.5.2
Using bundler 1.16.6
Using coffee-script-source 1.11.1
Using execjs 2.7.0
Using coffee-script 2.4.1
Using colorator 1.1.0
Using ruby-enum 0.7.2
Fetching commonmarker 0.17.11
Installing commonmarker 0.17.11 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory:
/Users/AlfieLuigi/railsgirls.github.io/vendor/bundle/ruby/2.5.0/gems/commonmarker-0.17.11/ext/commonmarker
/usr/local/opt/ruby/bin/ruby -r ./siteconf20181023-6866-4jwpvd.rb extconf.rb
creating Makefile

current directory:
/Users/AlfieLuigi/railsgirls.github.io/vendor/bundle/ruby/2.5.0/gems/commonmarker-0.17.11/ext/commonmarker
make "DESTDIR=" clean
xcrun: error: invalid active developer path
(/Library/Developer/CommandLineTools), missing xcrun at:
/Library/Developer/CommandLineTools/usr/bin/xcrun

current directory:
/Users/AlfieLuigi/railsgirls.github.io/vendor/bundle/ruby/2.5.0/gems/commonmarker-0.17.11/ext/commonmarker
make "DESTDIR="
xcrun: error: invalid active developer path
(/Library/Developer/CommandLineTools), missing xcrun at:
/Library/Developer/CommandLineTools/usr/bin/xcrun

make failed, exit code 1

Gem files will remain installed in
/Users/AlfieLuigi/railsgirls.github.io/vendor/bundle/ruby/2.5.0/gems/commonmarker-0.17.11
for inspection.
Results logged to
/Users/AlfieLuigi/railsgirls.github.io/vendor/bundle/ruby/2.5.0/extensions/x86_64-darwin-16/2.5.0/commonmarker-0.17.11/gem_make.out

An error occurred while installing commonmarker (0.17.11), and Bundler
cannot continue.
Make sure that `gem install commonmarker -v '0.17.11' --source
'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  github-pages was resolved to 191, which depends on
    jekyll-commonmark-ghpages was resolved to 0.1.5, which depends on
      jekyll-commonmark was resolved to 1.2.0, which depends on
        commonmarker
192-168-1-2:railsgirls.github.io AlfieLuigi$

**Comment by gjtorikian** [comment]
Does this solution help? https://apple.stackexchange.com/a/254381

**Comment by jasonzurita** [comment]
I am running into a similar problem... Have you found a solution to this @alfielytorres?

I did try that link @gjtorikian, any other suggestions?

**Comment by gjtorikian** [comment]
> I am running into a similar problem.

`similar != exact`. Could you post your stack trace here?

**Comment by jasonzurita** [comment]
You are totally right about that, sorry for the confusion! The symptom is that all ways I try to install commonmarker, it just hangs. Here is my stack trace from:
`gem install commonmarker -v '0.18.2' --source 'http://rubygems.org/' --verbose`

```
$ sudo gem install commonmarker -v '0.18.2' --source 'http://rubygems.org/' --verbose
HEAD https://api.rubygems.org/api/v1/dependencies
200 OK
HEAD http://api.rubygems.org/api/v1/dependencies
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=commonmarker
200 OK
GET http://api.rubygems.org/api/v1/dependencies?gems=commonmarker
200 OK
Getting SRV record failed: DNS result has no information for _rubygems._tcp.api.rubygems.org
GET https://api.rubygems.org/api/v1/dependencies?gems=ruby-enum
200 OK
GET http://api.rubygems.org/api/v1/dependencies?gems=ruby-enum
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=i18n
200 OK
GET http://api.rubygems.org/api/v1/dependencies?gems=i18n
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=concurrent-ruby
200 OK
GET http://api.rubygems.org/api/v1/dependencies?gems=concurrent-ruby
200 OK
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/LICENSE.txt
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/README.md
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/Rakefile
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/bin/commonmarker
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/commonmarker.gemspec
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/arena.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/autolink.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/autolink.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/blocks.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/buffer.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/buffer.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/case_fold_switch.inc
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/chunk.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/cmark-gfm-core-extensions.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/cmark-gfm-extension_api.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/cmark-gfm-extensions_export.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/cmark-gfm.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/cmark-gfm_export.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/cmark-gfm_version.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/cmark.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/cmark_ctype.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/cmark_ctype.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/commonmark.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/commonmarker.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/commonmarker.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/config.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/core-extensions.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/entities.inc
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/ext_scanners.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/ext_scanners.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/extconf.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/footnotes.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/footnotes.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/houdini.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/houdini_href_e.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/houdini_html_e.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/houdini_html_u.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/html.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/html.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/inlines.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/inlines.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/iterator.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/iterator.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/latex.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/linked_list.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/man.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/map.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/map.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/node.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/node.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/parser.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/plaintext.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/plugin.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/plugin.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/references.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/references.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/registry.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/registry.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/render.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/render.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/scanners.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/scanners.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/scanners.re
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/strikethrough.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/strikethrough.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/syntax_extension.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/syntax_extension.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/table.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/table.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/tagfilter.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/tagfilter.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/utf8.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/utf8.h
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker/xml.c
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/lib/commonmarker.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/lib/commonmarker/config.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/lib/commonmarker/node.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/lib/commonmarker/node/inspect.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/lib/commonmarker/renderer.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/lib/commonmarker/renderer/html_renderer.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/lib/commonmarker/version.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/benchmark.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/fixtures/curly.md
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/fixtures/dingus.md
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_attributes.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_basics.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_commonmark.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_doc.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_encoding.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_extensions.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_footnotes.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_gc.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_helper.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_linebreaks.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_maliciousness.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_node.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_options.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_pathological_inputs.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_plaintext.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_renderer.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_smartpunct.rb
/usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/test/test_spec.rb
Building native extensions. This could take a while...
current directory: /usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker
/usr/local/opt/ruby/bin/ruby -r ./siteconf20181205-15736-f3pbiz.rb extconf.rb
/usr/local/lib/ruby/site_ruby/2.5.0/rubygems/ext/builder.rb:72: warning: Insecure world writable dir /usr/local/lib/ruby/gems/2.5.0 in PATH, mode 040777
creating Makefile
current directory: /usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker
make "DESTDIR=" clean
current directory: /usr/local/lib/ruby/gems/2.5.0/gems/commonmarker-0.18.2/ext/commonmarker
make "DESTDIR="
```

And it just sits at the last line there.

I have tried:
- reinstalling cmake
- upgrading ruby to v2.5.3 (_ruby -v_ -> `ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-darwin18]`)
- installing earlier versions of commonmarker
- installing using `bundle install`
  + I am working on a jekyll site

**Comment by gjtorikian** [comment]
That is super strange!

I’m sorry for the added hassle, but would it be possible for you to clone this repo, run `bundle install`, and then try `bundle exec compile` to build the gem locally? Without any error I’m not at all certain what it could be, and I am hoping that with a local build an error message will be produced!

**Comment by jasonzurita** [comment]
No hassle! I appreciate your help with this.

I cloned the repo and tried to do a bundle install. Unfortunately, I got held up again at _Building native extensions. This could take a while..._ after _Installing github-markdown_. For some reason my system seems to be hanging up on the native extensions installation parts, so I don't think this is specific to commonmarker.

Thank you for your help! Now to figure out what is going on with _building native extensions_...
